client/src/components/inventory/HardwareList.tsx

```diff
@@ -54,97 +57,138 @@ const sampleData = [
-        });
-        const normalContent = (<Item.Group>
-            {sampleData.map((item) => (
-                <HardwareItem name={item.name}
-                              description={item.description}
-                              requestsEnabled={this.props.requestsEnabled && this.props.user}
-                              qtyRemaining={item.qtyRemaining}
-                              totalQty={item.totalQty}
-                              maxReqQty={item.maxReqQty}
-                              category={item.category}
-                              key={item.id}
-                              id={item.id}
-                              addItem={this.props.handleAddItem} // prop that invokes the handleAddItem method of parent container to update its state
-                              qtyUpdate={this.props.qtyUpdate} // this prop is the object whose request has been cancelled
-                              user={this.props.user}
-                />))}
-        </Item.Group>);
-        const loading = (<Item.Group>
-            <PlaceholderItem/>
-            <PlaceholderItem/>
-            <PlaceholderItem/>
-        </Item.Group>);
-
-        return (
-                <div>
+    public render() {
+        return <Query
+            query={gql`
+                query {
+                    items {
+                        id
+                        item_name
+                        description
+                        imageUrl
+                        category
+                        totalAvailable
+                        maxRequestQty
+                        hidden
+                        approvalRequired
+                        returnRequired
+                        owner
+                    }
+                }
+            `}>
+            {({loading, error, data}: any) => {
+                if (error) {
+                    console.log("error", error);
+                }
+                // TODO: come back to this
+                if (loading) {
+                    return (
+                        <div>
+                            <Header>Inventory</Header>
+                            const loadingGroup = (<Item.Group>
+                            <PlaceholderItem/>
+                            <PlaceholderItem/>
+                            <PlaceholderItem/>
+                        </Item.Group>);
+                        </div>
+                    );
+                }
+                if (error) {
+                    return <Message negative>
+                        <Message.Header>Error displaying hardware inventory</Message.Header>
+                        <p>Try refreshing the page. If that doesn't work, contact a member of the HackGT Team for
+                            assistance.</p>
+                    </Message>;
+                }
+
+                let noRequestsMessageText = "";
+
+                if (!this.props.requestsEnabled) {
```

Comment:
can you move these checks to outside of the query and only perform query if necessary?

Reply:
Didn't even think of it like that, great idea

Reply:
Wait, nevermind.  Even if requests are disabled, we can still show the hardware list -- it has a special mode that hides the request buttons and only shows item details.  However, I did refactor this code slightly so it's nicer now

Reply:
![image](https://user-images.githubusercontent.com/5790137/59153403-40a94d80-8a1e-11e9-8574-a1b22ded37c2.png)